crates/exex/src/model.rs

```diff
@@ -500,6 +539,59 @@ impl From<Header> for KethBlockHeader {
     }
 }
 
+/// [`KethPayload`] is a wrapper around a [`Vec<MaybeRelocatable>`], designed to represent
+/// a collection of `[MaybeRelocatable`] values in the Keth environment.
+///
+/// This struct is primarily used to encapsulate data that can injected into the Cairo VM for hint
+/// manipulation.
+#[derive(Debug, Eq, Ord, Hash, PartialEq, PartialOrd, Clone, Serialize, Deserialize)]
+pub struct KethPayload(pub Vec<MaybeRelocatable>);
+
+impl Deref for KethPayload {
+    type Target = Vec<MaybeRelocatable>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<KethBlockHeader> for KethPayload {
+    fn from(header: KethBlockHeader) -> Self {
+        let mut payload = Vec::new();
+
+        // Dynamically encode each field using a trait
+        let fields = [
+            &header.parent_hash as &dyn KethEncodable,
```

Comment:
why do we need `dyn` actually and cannot know at compile time that all fields types implement `KethEncodable`?

Reply:
Unfortunately this is impossible.

While all these types implement the `KethEncodable` trait, their underlying types are different, and Rust needs to know the exact type of each element at compile time when creating the array.